src/vs/workbench/contrib/chat/browser/contrib/chatInputEditorContrib.ts

```diff
@@ -131,7 +132,43 @@ class InputEditorDecorations extends Disposable {
 	}
 }
 
-ChatWidget.CONTRIBS.push(InputEditorDecorations);
+class InputEditorSlashCommandFollowups extends Disposable {
+	private slashCommands: Promise<ISlashCommand[] | undefined> | undefined;
+
+	constructor(
+		private readonly widget: IChatWidget,
+		@IChatService private readonly chatService: IChatService
+	) {
+		super();
+		this._register(this.chatService.onDidCompleteSlashCommand(({ slashCommand, sessionId }) => this.repopulateSlashCommand(slashCommand, sessionId)));
+	}
+
+	private getSlashCommands() {
+		if (!this.slashCommands && this.widget.viewModel?.sessionId) {
+			this.slashCommands = this.chatService.getSlashCommands(this.widget.viewModel?.sessionId, new CancellationTokenSource().token);
```

Comment:
When you have to do this, you can use `CancellationToken.None`, otherwise you leak a disposable